valhalla/thor/timedistancematrix.h

```diff
@@ -150,30 +98,21 @@ class TimeDistanceMatrix {
   // Edge status. Mark edges that are in adjacency list or settled.
   EdgeStatus edgestatus_;
 
-  AStarHeuristic astarheuristic_;
-
   sif::TravelMode mode_;
 
-  /**
-   * Expand from the node along the forward search path. Immediately expands
-   * from the end node of any transition edge (so no transition edges are added
-   * to the adjacency list or EdgeLabel list). Does not expand transition
-   * edges if from_transition is false.
-   * @param  graphreader  Graph tile reader.
-   * @param  node         Graph Id of the node being expanded.
-   * @param  pred         Predecessor edge label (for costing).
-   * @param  pred_idx     Predecessor index into the EdgeLabel list.
-   * @param  from_transition True if this method is called from a transition
-   *                         edge.
-   */
-  void ExpandForward(baldr::GraphReader& graphreader,
-                     const baldr::GraphId& node,
-                     const sif::EdgeLabel& pred,
-                     const uint32_t pred_idx,
-                     const bool from_transition);
+  template <const ExpansionType expansion_direction,
+            const bool FORWARD = expansion_direction == ExpansionType::forward>
+  std::vector<TimeDistance>
+  ComputeMatrix(const google::protobuf::RepeatedPtrField<valhalla::Location>& source_location_list,
+                const google::protobuf::RepeatedPtrField<valhalla::Location>& target_location_list,
+                baldr::GraphReader& graphreader,
+                const sif::mode_costing_t& mode_costing,
+                const sif::travel_mode_t mode,
+                const float max_matrix_distance,
+                const uint32_t matrix_locations = kAllLocations);
```

Comment:
this function's body is now what `SourceToTarget` was before, while `SourceToTarget` now only decides which direction to pick for the template. That way there's absolutely no change to calling functions.